autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClusterStatusSyncHandler.java

```diff
@@ -39,19 +40,22 @@ public void onApplicationEvent(ClusterStatusSyncEvent event) {
         }
         MDCBuilder.buildMdcContext(cluster);
 
-        Status cbClusterStatus = Optional.ofNullable(cloudbreakCommunicator
-                .getStackStatusByCrn(cluster.getStackCrn()).getClusterStatus()).orElse(Status.AMBIGUOUS);
-        LOGGER.debug("Analysing CBCluster Status '{}' for Cluster '{}' ", cbClusterStatus, cluster.getStackCrn());
+        StackStatusV4Response statusResponse = cloudbreakCommunicator.getStackStatusByCrn(cluster.getStackCrn());
+        boolean clusterAvailable = Optional.ofNullable(statusResponse.getStatus()).map(Status::isAvailable).orElse(false)
+                && Optional.ofNullable(statusResponse.getClusterStatus()).map(Status::isAvailable).orElse(false);
+        LOGGER.debug("Analysing CBCluster Status '{}' for Cluster '{}' ", statusResponse, cluster.getStackCrn());
 
-        if (DELETE_COMPLETED.equals(cbClusterStatus)) {
+        if (DELETE_COMPLETED.equals(statusResponse.getStatus())) {
             clusterService.removeById(autoscaleClusterId);
-            LOGGER.debug("Deleted cluster '{}', CB Cluster status '{}'.", cluster.getStackCrn(), cbClusterStatus);
-        } else if (cbClusterStatus.isAvailable() && !RUNNING.equals(cluster.getState())) {
+            LOGGER.info("Deleted cluster '{}', CB Stack Status '{}'.", cluster.getStackCrn(), statusResponse.getStatus());
+        } else if (clusterAvailable && !RUNNING.equals(cluster.getState())) {
             clusterService.setState(cluster.getId(), ClusterState.RUNNING);
-            LOGGER.debug("Updated cluster '{}' to running, CB Cluster status '{}'.", cluster.getStackCrn(), cbClusterStatus);
-        } else if (!cbClusterStatus.isAvailable() && RUNNING.equals(cluster.getState())) {
+            LOGGER.info("Updated cluster '{}' to Running, CB Stack Status '{}', CB Cluster Status '{}'.",
+                    cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+        } else if (!clusterAvailable && RUNNING.equals(cluster.getState())) {
             clusterService.setState(cluster.getId(), ClusterState.SUSPENDED);
-            LOGGER.debug("Suspended cluster '{}', CB Cluster status '{}'", cluster.getStackCrn(), cbClusterStatus);
+            LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'",
+                    cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
         }
```

Comment:
Periscope status should be synced based on both cluster and stack status, since next upscale is supported only when both are in available status. Even though they both are in sync in certain stimulated slow CB scenarios there is small drift between these two status.

Reply:
Thank you. Was going to ask just this question on how they differ.

Wasn't there a change recently where at least downscale worked even if a previous downscale had failed (or node deleted on provider side).
That's the series of jiras Amit had opened. cc @cegganesh84

Reply:
> Was going to ask just this question on how they differ.

stack -> represents the cloud, cluster -> represents the cm cluster. Yes, there can be drift. The first stack will be scaled to accommodate more nodes. Then the cluster will be scaled by the means of provisioning services. https://jira.cloudera.com/browse/CB-8051?focusedCommentId=2639540&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2639540. Extending this, will apply for downscale also in the reverse order.

Reply:
> Wasn't there a change recently where at least downscale worked even if a previous downscale had failed (or node deleted on provider side).

amit opened jiras were basically related to CB not identifying node deleted on provider side. It is not related to this change.